src/Amazon.Common.DotNetCli.Tools/PosixUserHelper.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+
+namespace Amazon.Common.DotNetCli.Tools
+{
+    public struct PosixUserInfo
+    {
+        public uint UserID;
+        public uint GroupID;
```

Comment:
You could also use nullable ints internally instead of checking `uint.MaxValue`. Although extremely unlikely, I guess technically a uid might be the max value.